targets/TARGET_Cypress/TARGET_PSOC6/mbed_overrides.c

```diff
@@ -0,0 +1,55 @@
+/*
+ * mbed Microcontroller Library
+ * Copyright (c) 2018-2019 Cypress Semiconductor Corporation
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "device.h"
+#include "psoc6_utils.h"
+#include "cycfg.h"
+
+#if defined(COMPONENT_SPM_MAILBOX)
+void mailbox_init(void);
+#endif
+
+/*******************************************************************************
+* Function Name: mbed_sdk_init
+****************************************************************************//**
+*
+* Mbed's post-memory-initialization function.
+* Used here to initialize common parts of the Cypress libraries.
+*
+*******************************************************************************/
+void mbed_sdk_init(void)
+{
+    /* Initialize shared resource manager */
+    cy_srm_initialize();
+
+    /* Initialize system and clocks. */
+    /* Placed here as it must be done after proper LIBC initialization. */
+    SystemInit();
+
+#if defined(COMPONENT_SPM_MAILBOX)
+    mailbox_init();
+#endif
+
+#if (!CY_CPU_CORTEX_M0P)
+    /* Set up the device based on configurator selections */
+    init_cycfg_all();
+
+    /* Enable global interrupts (disabled in CM4 startup assembly) */
+    __enable_irq();
```

Comment:
is this still needed ? Previously was disable-enable pair, now only enable in sdk_init

Reply:
Yes. As mentioned in the code comment, CM4 startup assembly masks the global interrupts in Reset_Handler (this behavior is required to enable correct startup sequence under all scenarios, the code is there since the initial PDL 3.0.0 release).
For example, see startup_psoc6_01_cm4.S: https://github.com/ARMmbed/mbed-os/blob/26aa16ff39df968757e2a71a480883bebf29c3c7/targets/TARGET_Cypress/TARGET_PSOC6/TARGET_CY8CKIT_062_WIFI_BT/device/TARGET_MCU_PSOC6_M4/TOOLCHAIN_GCC_ARM/startup_psoc6_01_cm4.S#L283